source/Nevermore/CommandParameterValues.cs

```diff
@@ -74,7 +74,29 @@ public void AddTable(string name, TableValuedParameter tvp)
         {
             Add(name, tvp);
         }
-        
+
+        static SqlMetaData GetSqlMetaData<T>(IReadOnlyCollection<T> ids, string name)
+        {
+            var valueType = typeof(T);
+            var notSupportedErrorMsg = $"'{valueType.Name}' is not a valid ID type, supported types are: {nameof(String)}, {nameof(Int32)}, {nameof(Int64)} and {nameof(Guid)}.";
+            switch (Type.GetTypeCode(valueType))
+            {
+                case TypeCode.String: return new SqlMetaData(name, SqlDbType.NVarChar, 300);
```

Comment:
Why did you select 300 instead of buckets as Paul suggested?

Reply:
`GetBestSizeBucket()`:
1. will use the wrong size if the length is longer than 200
2. the `ParameterValue` column can only be of length 300 at most now, using this method may give people the impression that we support any length
3. this suggestion was to help us optimize the execution plan if we wish to derive the length of the column from the id with the maximum length instead of specifying a fixed-length like 300

![image](https://user-images.githubusercontent.com/12688884/104268893-bae66180-54e0-11eb-9f0e-ab173024d27c.png)

Reply:
1: We could introduce a new 300 bucket. It does seem arbitrary. We'd need `ParamaterValue` to align with a bucket size?
2: `ParameterValue` can be adjusted by modifying the table, so Nevermore would support any value but provides a default of 300.
3: A fixed or bucket column length results in predictable query plans. The bucket suggestion was to keep the query plan optimisation while also allowing consumers of Nevermore to use a longer `ParameterValue` if they required it.

To me using the bucket seems like a more flexible option while retaining performance.

Reply:
We agreed to use bucket size here to gain flexibility and add some comment to explain our decision in code.

Reply:
hmm ... looks like it truncates the string value to an empty string if the `SqlMetaData` has a max length of 0.

![image](https://user-images.githubusercontent.com/12688884/104418053-14728d00-55c2-11eb-9049-34d763ac301a.png)

Let's catch up tmr morning before our daily sync.

Reply:
To Summarize:

We agreed to use a fixed length of 300 for `SqlMetaData` used for the `ParameterValue` column for now, which also aligns with the column length in our table schema in `Script0002-ParameterList.sql` due to time constraint.

We would probably revisit this issue later (perhaps during sharpening) if there is a need to dynamically resolve the max length based on each DocumentMap configuration.